product_readonly_security/models/product_mixin.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2024 Tecnativa - Víctor Martínez
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import _, api, models
+from odoo.exceptions import AccessError
+from odoo.tools import config
+
+
+class ProductReadonlySecurityMixin(models.AbstractModel):
+    _name = "product.readonly.security.mixin"
+    _description = "Mixin to use Product Readonly Security"
+
+    @api.model
+    def check_access_rights(self, operation, raise_exception=True):
+        """Simulate that you do not have ACLs so that the create, edit and delete
+        buttons are not displayed."""
```

Comment:
Don't overwrite the docstring, and remove the word "simulate", that makes it this as something fake, while this is totally legit. Also a bit more of explanation is needed:

```suggestion
        # Override security returning False/AccessError if not belonging to
        # the new security group. This makest that the create, edit and delete
        # buttons are not displayed.
```